src/GiraffeGenerator.Core/CodeGenNullChecking.fs

```diff
@@ -0,0 +1,212 @@
+﻿module CodeGenNullChecking
+
+open AST
+open ASTExt
+open FSharp.Compiler.SyntaxTree
+open OpenApi
+open OpenApiToAstTypesMatchingAndConversion
+
+type private PropertyPathSegment =
+  | Property of string
+  | NullableValue
+  | CollectionValue of PropertyPathSegment
+  | OptionValue of PropertyPathSegment
+
+let private isNullReference = "isNullReference"
+let private nullReferenceToOption = "nullReferenceToOption"
+let private defaultWithEmptySeq = Option.defaultValueExpr (longIdentExpr "Seq.empty")
+let private checkForUnexpectedNullsName = "checkForUnexpectedNulls"
+let generateNullCheckingHelpers () =
+    [
+        // isNullReference
+        letDecl false isNullReference ["v"] None // (=) doesn't work because F# thinks that C# can't fuck it up on nulls
```

Comment:
we have compiler intrinsic `isNull` btw

Reply:
which requires fsharp nullability of the type instance of which is checked. It would provide complier errors in generated code if value checked is a record, for example

Reply:
dropped null-checking as unstable and off-topic